receiver/statsdreceiver/internal/protocol/statsd_parser.go

```diff
@@ -430,6 +432,28 @@ func parseMessageToMetric(line string, enableMetricType bool, enableSimpleTags b
 
 				kvs = append(kvs, attribute.String(k, v))
 			}
+		case strings.HasPrefix(part, "c:"):
+			// As per DogStatD protocol v1.2:
+			// https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=metrics#dogstatsd-protocol-v12
+			containerID := strings.TrimPrefix(part, "c:")
+
+			if containerID != "" {
+				kvs = append(kvs, attribute.String(semconv.AttributeContainerID, containerID))
+			}
+		case strings.HasPrefix(part, "T"):
+			// As per DogStatD protocol v1.3:
+			// https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=metrics#dogstatsd-protocol-v13
```

Comment:
This is great, thanks for the comments. I didn't know about these extensions to statsd. I also am glad to see DD taking ownership and writing specs for their extensions to statsd!